massa-final-state/src/final_state.rs

```diff
@@ -77,9 +105,214 @@ impl FinalState {
             executed_ops,
             changes_history: Default::default(), // no changes in history
             final_state_hash: Hash::from_bytes(FINAL_STATE_HASH_INITIAL_BYTES),
+            last_start_period: 0,
         })
     }
 
+    /// Initializes a `FinalState` from a snapshot
+    ///
+    /// # Arguments
+    /// * `config`: the configuration of the execution state
+    pub fn from_snapshot(
```

Comment:
Done in eec62b33db173755bf5c609be58f15c018ee1728, let me know what you think about it :)

Reply:
Pretty happy with it.

There are a lot of clones of the config. Consider Rc-ing it, or using references instead of moves if this function could be a performance burden. If it's rarely called, leave as is

Reply:
It should never be called hopefully haha. Called once, on 1 node, for each emergency restart we have :)

Reply:
...where's the Oprah meme "you get a clone! you get a clone! everyone gets a clone!" when you need it!